orttraining/orttraining/core/graph/gradient_builder.cc

```diff
@@ -809,6 +809,30 @@ IMPLEMENT_GRADIENT_BUILDER(GetReduceSumGradient) {
   return result;
 }
 
+IMPLEMENT_GRADIENT_BUILDER(GetReduceSumTrainingGradient) {
```

Comment:
We don't need this at this moment... 
ReduceSumTraining should only be used in the gradient graph at this moment. 

When we updated the ONNX ReduceSum, we can update ReduceSumGradient accordingly...